### PR TITLE
[Task/cra 148 recreate] - Add Logging and logs file rotator

### PR DESCRIPTION
**Pull request by Riya-Singhal15**
Add Logging in each request and response and every function, and logs file rotator.

### CODE REVIEW

File: backend/.gitignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 package-lock.json
-info.log
-error.log
 .env
+logs/crs.log
```
Comment (nidhi-einn) [inline backend/.gitignore:3]: Backup logs file should also not added into git so logs should be created into a separate folder.

Comment (Riya-Singhal15) [reply backend/.gitignore]: Added log folder inside .gitignore file and committed the code (commit no. - 8a03c4a551c4e587b7499aac7816f93add738233)

File: backend/controller/examinationController.js [hunk 2]
```diff
@@ -147,27 +176,36 @@ const QuestionPut = function (req, res) {
 
 //Delete QuestionCollection
 const QuestionDelete = function (req, res) {
+  logger.info(req.url);
+  logger.info(req.headers);
+  logger.info(req.body);
```
Comment (nidhi-einn) [inline backend/controller/examinationController.js:181]: You are writing these 3 lines again and again can you create a function "printReq" to print req in detail.

File: backend/controller/examinationController.js [hunk 3]
```diff
@@ -176,13 +214,18 @@ const QuestionDelete = function (req, res) {
...
   if (req.session.user_type == Constants.admin) {
     //LETS VALIDATE THE DATA BEFORE WE ADD A PAPER
     const {
       error
     } = questionPaperValidation(req.body);
-    if (error) return res.status(400).send(error.details[0].message);
-
+    if (error) {
+      logger.error(errHandler.validationErrorHandler(error));
+      return res.status(400).send(error.details[0].message);
```
Comment (nidhi-einn) [inline backend/controller/examinationController.js:227]: Use constant instead of 400

File: backend/config/rotateLogAndDeleteOldLogFiles.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                    if (err) logger.error(errHandler.errorHandler(err));
+                    logger.info(`${findFileAtLocation}, file got removed`)
+                  });
+                }
+              }
+            });
+          }
+        }
+      });
+    }, Constants.oneDayInMilisecond); // every 5 hours
```
Comment (nidhi-einn) [inline backend/config/rotateLogAndDeleteOldLogFiles.js:98]: Change the comment

File: backend/config/rotateLogAndDeleteOldLogFiles.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+const logger = require("./logger");
+var dirPath = ["./logs/", "./androidLogs/"];
+
+function deleteOldLogFiles() {
+  setInterval(function () {
+    dirPath.forEach((directoryPath) => {
+      fs.readdir(directoryPath, (err, files) => {
+        if (err) logger.error(errHandler.errorHandler(err));
+        for (var i = 0; i < files.length; i++) {
+          if (files[i] == "crs.log") {
```
Comment (nidhi-einn) [inline backend/config/rotateLogAndDeleteOldLogFiles.js:14]: Use constants here

File: backend/controller/examinationController.js (hunk 2 above)
Comment (Riya-Singhal15) [reply backend/controller/examinationController.js]: Made function printLogs and printLogsWithBody in the logger.js file and committed the code (commit no. -dc917d258a99dba977571f8eb9884ebe499f0bda)

File: backend/config/rotateLogAndDeleteOldLogFiles.js (hunk 4 above)
Comment (Riya-Singhal15) [reply backend/config/rotateLogAndDeleteOldLogFiles.js]: Changed the comment and committed the code (commit no. -dc917d258a99dba977571f8eb9884ebe499f0bda)

File: backend/config/rotateLogAndDeleteOldLogFiles.js (hunk 5 above)
Comment (Riya-Singhal15) [reply backend/config/rotateLogAndDeleteOldLogFiles.js]: made a constant for "crs.log" and committed the code (commit no. -dc917d258a99dba977571f8eb9884ebe499f0bda)

File: backend/controller/examinationController.js (hunk 3 above)
Comment (Riya-Singhal15) [reply backend/controller/examinationController.js]: Changed the code and committed the code (commit no. -d7eddc9d06e4306e231b9ea3b430a068f9ec1778)

**Pull request merged**